catalogs/catalog-jdbc-common/src/main/java/com/datastrato/gravitino/catalog/jdbc/operation/JdbcTableOperations.java

```diff
@@ -106,7 +106,9 @@ public List<String> listTables(String databaseName) throws NoSuchSchemaException
       final List<String> names = Lists.newArrayList();
       try (ResultSet tables = getTables(connection)) {
         while (tables.next()) {
-          names.add(tables.getString("TABLE_NAME"));
+          if (Objects.equals(tables.getString("TABLE_SCHEM"), databaseName)) {
```

Comment:
Do we need to consider the case insensitive compare?

Reply:
Yes, but I'm not going to put the logic in the base class. Any JDBC catalog that is case-insensitive in the table name can overwrite this method. AFAIK, most JDBC tables are case-sensitive here.